src/router.ts

```diff
@@ -97,15 +99,24 @@ export class OneSchemaRouter<
 
   implement<Route extends keyof Schema & string>(
     route: Route,
-    implementation: EndpointImplementation<
-      Route,
-      z.output<Schema[Route]['request']>,
-      z.infer<Schema[Route]['response']>,
-      R
-    >,
-  ) {
+    ...middlewares: [
+      ...OneSchemaRouterMiddleware<
+        ContextOfEndpoint<Route, z.output<Schema[Route]['request']>, R>
+      >[],
+      EndpointImplementation<
+        Route,
+        z.output<Schema[Route]['request']>,
+        z.infer<Schema[Route]['response']>,
+        R
+      >,
+    ]
```

Comment:
So this above type for `middlewares` will require 0 to many `OneSchemaRouterMiddleware` parameters, followed by exactly 1 required `EndpointImplementation` parameter?

Reply:
@epeters3 Exactly! Previously, I thought this was impossible to model using TS. I was wrong :)

Reply:
Way cool. I think the fact that Typescript is fundamentally a correctness checker and DX tool and not just a necessary part of code compilation like with Java or C, has enabled the Typescript maintainers and community to do some pretty incredible things.